### PR TITLE
Add version tag support and increment version.

### PR DESCRIPTION
**Pull request by bakercp**
Reference https://github.com/openframeworks/openFrameworks/issues/3796

### CODE REVIEW

**Comment by arturoc** [comment]
i would call it master or nightly and then stable once we do a release

**Comment by bakercp** [comment]
Well, ideally (in my mind at least) for releases it will just be an empty string `""`.  Since we have moved away from the develop branch model I suppose it should be called "master" ... or in some future build-system it should be updated with build tools :)

I'll update the PR w/ master for now.

**Comment by bilderbuchi** [comment]
+1 on this, and thank you for making it fully semver-compatible.
Also, +1 on having this as an empty string for release versions, maybe this should be added as a comment, and as info in the release checklist on the wiki.
finally, maybe it's a good idea to not call this "TAG" but "PRERELEASE_INDICATOR" (or somethign shorter) to make the meaning clearer for people not intimately familiar with the semver specification (i.e. that hyphenated postfixes indicate prerelease versions).

**Comment by bakercp** [comment]
@bilderbuchi I see you've answered my question over here.  Regarding what to call it -- I can't seem to find a clear "spec" answer ... i've seen patch-extension and seen it referred to as a tag ... (older versions of the spec seem to call it tag).

Anyway, I'd prefer to follow the semver2.0.0 naming convention if we can figure out what that is ...

**Comment by bilderbuchi** [comment]
I think the spec only details the contents of the version string, and not the "name" of the parts making it up, but if you follow the system of the spec items 6 to 9, `PRERELEASE` (or more precisely `PRE-RELEASE`, but that seems strange for a preprocessor variable) seems to be most consistent with the semver specification.

**Comment by bakercp** [comment]
@bilderbuchi I think you nailed the name -- it should indeed be a version of pre-release. I'd suggest `OF_VERSION_PRE_RELEASE`.  I did take a look to see if we had any other examples of collapsing hyphens/dashes in variable names (constants) in oF and I can't seem to find any.  It comes down to an aesthetic thing for me since we can't really lean on consistency and .. for whatever reason, even though it's long, I find PRE_RELEASE easier to read.

**Comment by bilderbuchi** [comment]
ok, let's do that, then. Incidentally, it also makes the meaning clear without any additional comment or documentation.

**Comment by bakercp** [comment]
As soon as this is merged, I can update the wiki checklist https://github.com/openframeworks/openFrameworks/wiki/Release-preparation-checklist

File: libs/openFrameworks/utils/ofUtils.h [hunk 1]
```diff
@@ -820,6 +820,20 @@ unsigned int ofGetVersionMinor();
 /// \returns The patch version number.
 unsigned int ofGetVersionPatch();
 
+/// \brief Get the pre-release version of openFrameworks.
+///
+/// openFrameworks uses the semantic versioning system.
+///
+/// For pre-release versions of openFrameworks, including development versions,
+/// this string will describe the pre-release state. Examples might include
+/// "master", "rc1", "rc2", etc.  For all official releases, this string will be
```
Comment (bilderbuchi) [inline libs/openFrameworks/utils/ofUtils.h]: maybe replace "official" with "full" or "proper"/"main" - prereleases are also official in that we (should) decide which commit is the prerelease state, and set a git tag, etc.

Comment (bakercp) [inline libs/openFrameworks/utils/ofUtils.h]: Ah good point.  "For all stable releases, ..."

File: libs/openFrameworks/utils/ofUtils.cpp [hunk 2]
```diff
@@ -737,7 +737,14 @@ void ofLaunchBrowser(const string& _url, bool uriEncodeQuery){
 //--------------------------------------------------
 string ofGetVersionInfo(){
 	stringstream sstr;
-	sstr << OF_VERSION_MAJOR << "." << OF_VERSION_MINOR << "." << OF_VERSION_PATCH << endl;
+	sstr << OF_VERSION_MAJOR << "." << OF_VERSION_MINOR << "." << OF_VERSION_PATCH;
+
+	if (sizeof(OF_VERSION_PRE_RELEASE) > 1)
```
Comment (arturoc) [inline libs/openFrameworks/utils/ofUtils.cpp]: using sizeof here is kind of weird no? something like `string(OF_VERSION_PRE_RELEASE).length()` makes more sense i think

**Comment by bakercp** [comment]
@arturoc  -- fixed that sizeof thing.  Initially I was having problems with string literals in preprocessor defines and found that solution on stackoverflow.  Clearly there was a different problem, so I fixed it using a string.  Much prettier now.  Good to merge?

**Comment by arturoc** [comment]
thanks!

**Comment by bakercp** [comment]
+1 :)

**Comment by bakercp** [comment]
@dimitre - master now reports a more useful version string.  CB

**Comment by armadillu** [comment]
Thx guys! :)

**Comment by dimitre** [comment]
pure bliss, thank you guys! :)

**Pull request merged**